tools/missing-test-detector/changed.go

```diff
@@ -7,20 +7,30 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-// FieldCoverage is a nested map with field names as keys and whether the field is covered as values.
+// ResourceChanges is a nested map with field names as keys and Field objects
+// as bottom-level values.
 // Fields are assumed not to be covered until detected in a test.
-type FieldCoverage map[string]any
+type ResourceChanges map[string]any
+
+type Field struct {
+	// Added is true when the field is newly added between oldProvider and newProvider.
+	Added bool
+	// Changed is true when the field type has changed between oldProvider and newProvider.
+	Changed bool
+	// Tested is true when a test has been found that includes the field.
+	Tested bool
```

Comment:
Not a blocker, but in case it becomes relevant / would be easy to fit in to future refactors: in the future we will likely want to fully separate the diff computation from the check whether the fields in the diff are tested.

Reply:
I'm struggling to find a way to fit this into this PR.

Reply:
👍 it's fine to leave out of this PR.